### ISSUE TITLE
`PedersenCommitment` blackbox function should allow the point at infinity as an output

### ISSUE THREAD

**Issue by TomAFrench** [noir-lang/noir#5196]
Consider the code:

```rust
 let pedersen_commitment_to_zero = std::hash::pedersen_commitment([0]);
println(pedersen_commitment_to_zero);
```
executing this code gets the panic

```
% nargo execute   ~/noir/test_programs/execution_success/pedersen_check gd/issue_4931 + tom-box
The application panicked (crashed).
Message:  should not commit to point at infinity
Location: acvm-repo/bn254_blackbox_solver/src/lib.rs:48
```

This is because we assume that the pedersen commitment returns a point which lies on the curve 
https://github.com/noir-lang/noir/blob/070d7e71d6587679721437dd5b996478bd17ed85/acvm-repo/bn254_blackbox_solver/src/lib.rs#L47-L50

We should update this blackbox function so it has 3 return witnesses so that we can represent the point at infinity.

**Comment by guipublic** [comment]
This is resolved by #5221

**Comment by TomAFrench** [comment]
this opcode no longer exists.
